src/types/viewport.cpp

```diff
@@ -848,10 +848,10 @@ Viewport Viewport::ToOrigin() const noexcept
     SHORT newRight = original._sr.Right;
     SHORT newBottom = original._sr.Bottom;
 
-    THROW_IF_FAILED(ShortAdd(newLeft, delta.X, &newLeft));
-    THROW_IF_FAILED(ShortAdd(newRight, delta.X, &newRight));
-    THROW_IF_FAILED(ShortAdd(newTop, delta.Y, &newTop));
-    THROW_IF_FAILED(ShortAdd(newBottom, delta.Y, &newBottom));
+    THROW_HR_IF(E_ABORT, !base::CheckAdd(newLeft, delta.X).AssignIfValid(&newLeft));
+    THROW_HR_IF(E_ABORT, !base::CheckAdd(newRight, delta.X).AssignIfValid(&newRight));
+    THROW_HR_IF(E_ABORT, !base::CheckAdd(newTop, delta.Y).AssignIfValid(&newTop));
+    THROW_HR_IF(E_ABORT, !base::CheckAdd(newBottom, delta.Y).AssignIfValid(&newBottom));

```

Comment:
Shouldn't these `E_ABORT` errors be `INTSAFE_E_ARITHMETIC_OVERFLOW` if we want to remain compatible with the original code? It's probably not a big deal, but it's theoretically possible someone could be checking for specific error returns in some situation, especially if the return code is exposed in a public API somewhere.

Reply:
Although I suppose if we want to get rid of the _intsafe.h_ dependency then I guess that's not an option, unless you want to redefine the constant, or hardcode the error number. I don't know.

Reply:
Hum. You're right that it probably should maintain the error code in case anyone's checking specifically for it.... but I also highly doubt that anyone is given that I probably introduced it in the last few years and most things I've ever found are only checking overall success/failure from any of our API returns (if they even check at all and don't just steamroll anyway).

If we're worried about it, I don't mind redefining that constant without the header if the goal is to remove the header. But also, I don't super mind if it just becomes a different error code either...

Reply:
Yeah OK. The more I think about it, the more I think it's probably not a big deal, especially if this code was only a fairly recent addition.